amads/melody/contour/interpolation_contour.py

```diff
@@ -0,0 +1,272 @@
+            next_pitch = pitches[i + 1]
+
+            # Skip changing notes (where adjacent pitches are equal)
+            if prev_pitch == next_pitch:
+                continue
+
+            # Check if it's a peak or valley
+            if (prev_pitch < curr_pitch and next_pitch < curr_pitch) or (
+                prev_pitch > curr_pitch and next_pitch > curr_pitch
+            ):
+                # Additional check for equal adjacent notes
+                is_extremum = True
+                if i > 1 and prev_pitch == curr_pitch:
+                    is_extremum = (pitches[i - 2] - curr_pitch) * (
+                        next_pitch - curr_pitch
+                    ) < 0
+                if i < n - 2 and next_pitch == curr_pitch:
+                    is_extremum = (prev_pitch - curr_pitch) * (
+                        pitches[i + 2] - curr_pitch
+                    ) < 0
+                if is_extremum:
+                    extrema_indices.append(i)
+
+        extrema_indices.append(n - 1)  # Add last note
+        return np.array(extrema_indices)
+
+    def calculate_interpolation_contour(self) -> list[float]:
+        """Calculate the interpolation contour [1] representation of a melody.
+
+        Returns
+        -------
+        list[float]
+            Array containing the interpolation contour representation
+
+        References
+        ----------
+        [1] Müllensiefen, D. (2009). FANTASTIC: Feature ANalysis Technology Accessing
+        STatistics (In a Corpus): Technical Report v1.5
+        https://www.doc.gold.ac.uk/isms/m4s/FANTASTIC_docs.pdf
+        """
+        # Step 1 & 2: Find contour extrema
+        extrema_indices = self._find_contour_extrema(self.pitches)
+        extrema_times = [self.times[i] for i in extrema_indices]
+        extrema_pitches = [self.pitches[i] for i in extrema_indices]
+
+        # Step 3: Calculate gradients
+        gradients = np.diff(extrema_pitches) / np.diff(extrema_times)
+
+        # Step 4: Calculate durations
+        durations = np.diff(extrema_times)
+
+        # Step 5: Remove durations below 0.05 seconds
+        durations[durations < 0.05] = 0
+
+        # Step 6: Create weighted gradients vector by repeating each gradient
+        # proportional to its duration
+        # Convert durations to integer number of samples (at 100Hz sampling rate)
+        samples_per_duration = np.round(durations * 100).astype(int)
+        interpolation_contour = np.repeat(gradients, samples_per_duration)
+        return [float(x) for x in interpolation_contour]
+
+    @property
+    def direction(self) -> list[int]:
+        """Calculate the global direction of the interpolation contour.
+
+        Returns
+        -------
+        list[int]
+            List containing the signs (+1, 0, -1) of the gradient values in the contour
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.direction[:4]  # Show first 4 values
```

Comment:
This is a rather incomplete test that does not illustrate the full functionality!